test/boost/stall_free_test.cc

```diff
@@ -53,3 +55,523 @@ SEASTAR_THREAD_TEST_CASE(test_merge4) {
+    std::array<std::unique_ptr<X>, count> a;
+    int cleared_gently = 0;
+
+    for (int i = 0; i < count; i++) {
+        a[i] = std::make_unique<X>(i, [&cleared_gently] (int) {
+            cleared_gently++;
+        });
+    }
+
+    utils::clear_gently(a).get();
+    BOOST_REQUIRE_EQUAL(cleared_gently, count);
+}
+
+SEASTAR_THREAD_TEST_CASE(test_clear_gently_trivial_vector) {
+    constexpr int count = 100;
+    std::vector<int> v;
+
+    v.reserve(count);
+    for (int i = 0; i < count; i++) {
+        v.emplace_back(i);
+    }
+
+    utils::clear_gently(v).get();
+    BOOST_CHECK(v.empty());
+}
+
+SEASTAR_THREAD_TEST_CASE(test_clear_gently_trivial_small_vector) {
+    utils::small_vector<int, 1> v;
+    constexpr int count = 10;
+
+    v.reserve(count);
+    for (int i = 0; i < count; i++) {
+        v.emplace_back(i);
+    }
+
+    utils::clear_gently(v).get();
+    BOOST_CHECK(v.empty());
+}
+
+SEASTAR_THREAD_TEST_CASE(test_clear_gently_vector) {
+    static thread_local std::vector<int> res;
+    struct X {
+        int v;
+        X(int i) : v(i) {}
+        X(X&& x) noexcept : v(x.v) {}
+        future<> clear_gently() noexcept {
+            res.push_back(v);
+            return make_ready_future<>();
+        }
+    };
+    static_assert(std::is_trivially_destructible_v<X>);
+    std::vector<X> v;
+    constexpr int count = 100;
+
+    res.reserve(count);
+    v.reserve(count);
+    for (int i = 0; i < count; i++) {
+        v.emplace_back(X(i));
+    }
+
+    utils::clear_gently(v).get();
+    BOOST_CHECK(v.empty());
+
+    // verify that the items were cleared in reverse order
+    for (int i = 0; i < count; i++) {
+        BOOST_REQUIRE_EQUAL(res[i], 99 - i);
+    }
+}
+
+SEASTAR_THREAD_TEST_CASE(test_clear_gently_small_vector) {
+    std::vector<int> res;
+    struct X {
+        std::unique_ptr<int> v;
+        std::function<void (int)> on_clear;
+        X(int i, std::function<void (int)> f) : v(std::make_unique<int>(i)), on_clear(std::move(f)) {}
+        X(X&& x) noexcept : v(std::move(x.v)), on_clear(std::move(x.on_clear)) {}
+        X& operator=(X&& x) noexcept {
+            if (&x != this) {
+                std::swap(v, x.v);
+                std::swap(on_clear, x.on_clear);
+            }
+            return *this;
+        }
+        future<> clear_gently() noexcept {
+            on_clear(*v);
+            v.reset();
+            return make_ready_future<>();
+        }
+    };
+    utils::small_vector<X, 1> v;
+    constexpr int count = 100;
+
+    res.reserve(count);
+    v.reserve(count);
+    for (int i = 0; i < count; i++) {
+        v.emplace_back(X(i, [&res] (int i) {
+            res.emplace_back(i);
+        }));
+    }
+
+    utils::clear_gently(v).get();
+    BOOST_CHECK(v.empty());
+
+    // verify that the items were cleared in reverse order
+    for (int i = 0; i < count; i++) {
+        BOOST_REQUIRE_EQUAL(res[i], 99 - i);
+    }
+}
+
+SEASTAR_THREAD_TEST_CASE(test_clear_gently_chunked_vector) {
```

Comment:
Just a thought - you could have used a template to test vector, chunked_vector, deque and a bunch of other types as you chose - without duplicating the test code.

Reply:
True.  I thought of that, but then it is harder to tell which case failed if a BOOST_REQUIRE fails in the templated code.